modules/peridynamics/src/materials/ComputeStrainBaseNOSPD.C

```diff
@@ -77,7 +84,75 @@ ComputeStrainBaseNOSPD::computeQpDeformationGradient()
   _ddgraddv[_qp].zero();
   _ddgraddw[_qp].zero();
   _multi[_qp] = 0.0;
+  _sf_coeff[_qp] = 0.0;
+
+  if (_dim == 2)
+    _shape2[_qp](2, 2) = _deformation_gradient[_qp](2, 2) = 1.0;
+
+  if (_stabilization == "FORCE")
+    computeConventionalQpDeformationGradient();
+  else if (_stabilization == "HORIZON")
+    computeBondHorizonQpDeformationGradient();
+  else
+    paramError("stabilization",
+               "Unknown stabilization scheme for peridynamic correspondence model. Choose from: "
```

Comment:
I'd drop the "Choose from" sentence. If you were to add another option, it won't show up in the list. Plus, you shouldn't even be able to get to that point in the code.